client/utils/guidance.py

```diff
@@ -349,3 +367,25 @@ def set_tg(self):
 
         tg_file = open(self.current_working_dir + LOCAL_TG_FILE, 'r')
         self.tg_id = int(tg_file.read())
+
+    def prompt_with_prob(self, orig_response=None, prob=None):
+        """Ask for rationale with a specific level of probability. """
+        if self.assignment.cmd_args.no_hints:
```

Comment:
Not quite the best flag - but I don't want to add a new one. 

Having an opt-out for explanations is a good idea